python/cuspatial/cuspatial/tests/binpreds/test_binpred_test_dispatch.py

```diff
@@ -71,75 +61,103 @@ def test_simple_features(
         diagnostic output.
     """
-    try:
+    if not LOG_DISPATCHED_PREDICATES:
         (lhs, rhs) = simple_test[2], simple_test[3]
         gpdlhs = lhs.to_geopandas()
         gpdrhs = rhs.to_geopandas()
+
+        # Reverse
+        pred_fn = getattr(rhs, predicate)
+        got = pred_fn(lhs)
+        gpd_pred_fn = getattr(gpdrhs, predicate)
+        expected = gpd_pred_fn(gpdlhs)
+        assert (got.values_host == expected.values).all()
+
+        # Forward
         pred_fn = getattr(lhs, predicate)
         got = pred_fn(rhs)
         gpd_pred_fn = getattr(gpdlhs, predicate)
         expected = gpd_pred_fn(gpdrhs)
         assert (got.values_host == expected.values).all()
-
-        # The test is complete, the rest is just logging.
+    else:
         try:
-            # The test passed, store the results.
-            predicate_passes[predicate] = (
+            (lhs, rhs) = simple_test[2], simple_test[3]
+            gpdlhs = lhs.to_geopandas()
+            gpdrhs = rhs.to_geopandas()
+
+            # Reverse
+            pred_fn = getattr(rhs, predicate)
+            got = pred_fn(lhs)
+            gpd_pred_fn = getattr(gpdrhs, predicate)
+            expected = gpd_pred_fn(gpdlhs)
+            assert (got.values_host == expected.values).all()
+
+            # Forward
+            pred_fn = getattr(lhs, predicate)
+            got = pred_fn(rhs)
+            gpd_pred_fn = getattr(gpdlhs, predicate)
+            expected = gpd_pred_fn(gpdrhs)
+            assert (got.values_host == expected.values).all()
+
+            # The test is complete, the rest is just logging.
+            try:
+                # The test passed, store the results.
+                predicate_passes[predicate] = (
+                    1
+                    if predicate not in predicate_passes
+                    else predicate_passes[predicate] + 1
+                )
+                feature_passes[(lhs.column_type, rhs.column_type)] = (
+                    1
+                    if (lhs.column_type, rhs.column_type) not in feature_passes
+                    else feature_passes[(lhs.column_type, rhs.column_type)] + 1
+                )
+                passes_df = pd.DataFrame(
+                    {
+                        "predicate": list(predicate_passes.keys()),
+                        "predicate_passes": list(predicate_passes.values()),
+                    }
+                )
+                passes_df.to_csv("predicate_passes.csv", index=False)
+                passes_df = pd.DataFrame(
+                    {
+                        "feature": list(feature_passes.keys()),
+                        "feature_passes": list(feature_passes.values()),
+                    }
+                )
+                passes_df.to_csv("feature_passes.csv", index=False)
+            except Exception as e:
+                raise ValueError(e)
+        except Exception as e:
+            # The test failed, store the results.
+            out_file.write(
```

Comment:
In this case the `except` on line 131 is the handler for the `try:` on line 83. I think that if the test throws it will resolve normal error handling if LOG_DISPATCH_PREDICATES is not defined, right?